packages/collections/graphql/resolvers/User.js

```diff
@@ -23,5 +24,22 @@ module.exports = {
     ) {
       return Collection.byNameForUser(name, user.id, context)
     }
+  },
+  async collectionItems (user, args, context) {
+    const { user: me } = context
+    if (
+      (Roles.userIsMe(user, me) && Roles.userIsInRoles(user, accessRoles)) ||
+      Roles.userIsInRoles(me, adminRoles)
+    ) {
+      const collections = await Promise.all(args.names.map(name => Collection.byNameForUser(name, user.id, context)))
```

Comment:
We could reduce the amount of call to the DB as follows, right?
```suggestion
       const collections = await Collection.findForUser(user.id, context)
      .then(collections => collections.filter(
        collection => args.names.includes(collection.name)
      ))
```

This code was not tested ;)

Reply:
`Collection.byNameForUser` uses a data loader under the hood which gets those collections in one sql query. So the current implementation is more efficient because it only loads the ones that are requested and also only does one db query.